applications/StructuralMechanicsApplication/python_scripts/symbolic_generation/total_lagrangian_mixed_volumetric_strain_element/total_lagrangian_mixed_volumetric_strain_element_template.cpp

```diff
@@ -0,0 +1,1261 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:         BSD License
+//                   license: structural_mechanics_application/license.txt
+//
+//  Main authors:    Ruben Zorrilla
+//                   Riccardo Rossi
+//                   Guglielmo Scovazzi
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "utilities/element_size_calculator.h"
+#include "utilities/geometry_utilities.h"
+#include "utilities/math_utils.h"
+#include "includes/checks.h"
+
+// Application includes
+#include "custom_elements/total_lagrangian_mixed_volumetric_strain_element.h"
+#include "custom_utilities/structural_mechanics_element_utilities.h"
+
+namespace Kratos
+{
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<>
+double TotalLagrangianMixedVolumetricStrainElement<2>::CalculateShearModulus(const Matrix &rC) const
+{
+    return 0.2*(rC(0,0) - 2.0*rC(0,1) + rC(1,1) + rC(2,2));
```

Comment:
This can be replaced with if constexpr if you want

Reply:
Agree. I created this before we moved to the c++17. I'll change it.

Reply:
Indeed, after having a look to the code it makes no sense at all in this case as this is called from an already specialized function, which should be always there as it is the one containing the AD result.